app/controllers/gigs_controller.rb

```diff
@@ -0,0 +1,65 @@
+class GigsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :require_admin, only: [ :manage, :new, :create, :edit, :destroy ]
+  before_action :set_gig,       only: [ :show, :edit, :update ]
+
+  def new
+    @gig = Gig.new
+  end
+
+  def create
+    @gig = Gig.new(allowed_gig_params)
+    if @gig.save
+      flash[:notice] = "New gig added. #{total_msg}"
+      #set_flash_message :notice, :new_gig_added, count: Gig.count
+      redirect_to @gig
+    else
+      render :new
+    end
+  end
+
+  def show
+  end
+
+  def manage
+    @gigs = Gig.all
+  end
+
+  def edit
+  end
+
+  def update
+    if @gig.update(allowed_gig_params)
+      redirect_to gig_path(params[:id])
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    Gig.find(params[:id]).destroy
+    flash[:notice] = "gig deleted.  #{total_msg}"
+    redirect_to manage_gigs_path()
+  end
+
+  private
+
+  def set_gig
+    @gig = Gig.find(params[:id])
```

Comment:
This took me a while to internalize, but it is a good way to do this and simplifies the views that use `gig` somewhat as well.